beacon-chain/p2p/rpc_topic_mappings.go

```diff
@@ -54,3 +96,94 @@ func VerifyTopicMapping(topic string, msg interface{}) error {
 	}
 	return nil
 }
+
+// TopicDeconstructor splits the provided topic to its logical sub-sections.
+// It is assumed all input topics will follow the specific schema:
+// /protocol-prefix/message-name/schema-version/...
+// For the purposes of deconstruction, only the first 3 components are
+// relevant.
+func TopicDeconstructor(topic string) (string, string, string, error) {
+	origTopic := topic
+	protPrefix := ""
+	message := ""
+	version := ""
+
+	// Iterate through all the relevant mappings to find the relevant prefixes,messages
+	// and version for this topic.
+	for k := range protocolMapping {
+		keyLen := len(k)
+		if keyLen > len(topic) {
+			continue
+		}
+		if topic[:keyLen] == k {
+			protPrefix = k
+			topic = topic[keyLen:]
+		}
+	}
+
+	if protPrefix == "" {
+		return "", "", "", errors.Errorf("unable to find a valid protocol prefix for %s", origTopic)
+	}
+
+	for k := range messageMapping {
+		keyLen := len(k)
+		if keyLen > len(topic) {
+			continue
+		}
+		if topic[:keyLen] == k {
+			message = k
+			topic = topic[keyLen:]
+		}
+	}
+
+	if message == "" {
+		return "", "", "", errors.Errorf("unable to find a valid message for %s", origTopic)
+	}
+
+	for k := range versionMapping {
+		keyLen := len(k)
+		if keyLen > len(topic) {
+			continue
+		}
+		if topic[:keyLen] == k {
+			version = k
+			topic = topic[keyLen:]
+		}
+	}
+
+	if version == "" {
+		return "", "", "", errors.Errorf("unable to find a valid schema version for %s", origTopic)
+	}
+
+	return protPrefix, message, version, nil
+}
+
+// RPCTopic is a type used to denote and represent a req/resp topic.
+type RPCTopic string
+
+// ProtocolPrefix returns the protocol prefix of the rpc topic.
+func (r RPCTopic) ProtocolPrefix() string {
+	prefix, _, _, err := TopicDeconstructor(string(r))
+	if err != nil {
+		return ""
```

Comment:
`TopicDeconstructor` will actually return the error but this suppresses the error. Seems counter productive?

Reply:
Wanted to avoid having to return an error so as to make usage simpler. We can handle the empry string returned by the caller. It it still seems off, i can return an error instead.

Reply:
I prefer simpler as well